src/cpp/StatisticsBackend.cpp

```diff
@@ -182,11 +179,19 @@ EntityId create_and_register_monitor(
         const DomainParticipantQos& participant_qos,
         const DomainId domain_id = 0)
 {
+    // NOTE: This method is quite awful to read because of the error handle of every entity
```

Comment:
Error handling could be done with [scope_exit](https://en.cppreference.com/w/cpp/experimental/scope_exit), but it is C++20. We can have our own implementation, for instance inspired on [this one](https://github.com/ros2/rcpputils/blob/rolling/include/rcpputils/scope_exit.hpp)